crates/trident/src/subsystems/network.rs

```diff
@@ -41,9 +42,22 @@ impl Subsystem for NetworkSubsystem {
 }
 
 fn disable_cloud_init_networking() -> Result<(), TridentError> {
-    fs::write(CLOUD_INIT_DISABLE_FILE, CLOUD_INIT_DISABLE_CONTENT)
+    let cloud_init_disable_path = Path::new(CLOUD_INIT_CONFIG_DIR).join(CLOUD_INIT_DISABLE_FILE);
+    if !Path::new(CLOUD_INIT_CONFIG_DIR).exists() {
```

Comment:
I think this is sufficient to address this bug, but should we have a better handling re package presence detection in general? We could use the cosi manifest now, right?